src/main/kotlin/kotlinx/kover/adapters/KotlinMultiplatformPluginAdapter.kt

```diff
@@ -4,29 +4,43 @@
 
 package kotlinx.kover.adapters
 
+import groovy.lang.*
 import kotlinx.kover.adapters.api.*
 import org.gradle.api.*
-import org.jetbrains.kotlin.gradle.dsl.*
-import org.jetbrains.kotlin.gradle.plugin.*
+import org.gradle.api.file.*
+import org.gradle.internal.metaobject.*
 
 class KotlinMultiplatformPluginAdapter : CompilationPluginAdapter {
 
+    @Suppress("UNCHECKED_CAST")
     override fun findDirs(project: Project): PluginDirs {
         return safe(project) {
             this.plugins.findPlugin("kotlin-multiplatform") ?: return@safe PluginDirs(emptyList(), emptyList())
 
-            val extension = project.extensions.findByType(
-                KotlinMultiplatformExtension::class.java
-            ) ?: return@safe PluginDirs(emptyList(), emptyList())
+            val extension =
```

Comment:
This part of the trickery definitely needs some explanation with comments for the sake of future maintainability

Reply:
+1

Reply:
@qwwdfsad, @sandwwraith In addition to the comment, I can move reflection to the fail safe path, because in most cases the Kover will be declared in the same project as the kotlin Multiplatform Plugin.

It will be a copy paste, but it will work faster and there will be an example of exactly how the code should work with the reflection.

Reply:
Yeah, that would be nice